include/crpropa/module/PhotoPionProduction.h

```diff
@@ -60,6 +61,17 @@ class PhotoPionProduction: public Module {
 	 @param z		redshift
 	 */
 	double lossLength(int id, double gamma, double z = 0);
+
+	/**
+	 Direct SOPHIA interface.
+	 Output is a vector of length 2*N where N is the number of out-going particles.
```

Comment:
Is there a reason not to define a simple 
struct SophiaParticle{
int id;
double energy;
};
Storing the particle data in the double array is really ugly.